src/Adapter/Module/Tab/ModuleTabRegister.php

```diff
@@ -0,0 +1,237 @@
+    }
+    
+    /**
+     * Check mandatory data for tab registration, such as class name and class exists
+     * 
+     * @param string $moduleName
+     * @param ParameterBag $data
+     * @return boolean (= true) when no issue detected
+     * @throws Exception in case of invalid data
+     */
+    protected function checkIsValid($moduleName, ParameterBag $data)
+    {
+        $className = $data->get('class_name', null);
+        if (null === $className) {
+            throw new Exception('Missing class name of tab');
+        }
+        // Check controller exists
+        if (!in_array($className.'Controller.php', $this->getModuleAdminControllersFilename($moduleName))) {
+            throw new Exception(sprintf('Class "%sController" not found in controllers/admin', $className));
+        }
+        return true;
+    }
+    
+    /**
+     * Find all ModuleAdminController classes from module.
+     * This allow to check a class exists for a registered tab and to register automatically all the classes
+     * not explicitely declared by the module developer.
+     * 
+     * @param string $moduleName
+     * @return array of Symfony\Component\Finder\SplFileInfo, listing all the ModuleAdminControllers found
+     */
+    protected function getModuleAdminControllers($moduleName)
+    {
+        $modulePath = _PS_ROOT_DIR_.'/'.basename(_PS_MODULE_DIR_).
+                '/'.$moduleName.'/controllers/admin/';
+        
+        if (!$this->filesystem->exists($modulePath)) {
+            return array();
+        }
+        
+        $moduleFolder = $this->finder->files()
+                    ->in($modulePath)
+                    ->depth('== 0')
+                    ->name('*.php')
+                    ->exclude(['index.php'])
+                    ->contains('/extends\s+ModuleAdminController/i');
+        
+        return iterator_to_array($moduleFolder);
+    }
+    
+    /**
+     * Convert SPLFileInfo array to file names. Better & easier to check if a class to register exists.
+     * 
+     * @param string $moduleName
+     * @return array of strings
+     */
+    protected function getModuleAdminControllersFilename($moduleName)
+    {
+        return array_map(function(SplFileInfo $file) {
+            return $file->getFilename();
+        }, $this->getModuleAdminControllers($moduleName));
+    }
+    
+    /**
+     * Install a tab according to its defined structure
+     *
+     * @param Module $module
+     * @param ParameterBag $data The structure of the tab.
+     *
+     * @throws Exception in case of error from validation or save
+     */
+    protected function registerTab(Module $module, ParameterBag $data)
+    {
+        $this->checkIsValid($module->get('name'), $data);
+        
+        // Legacy Tab, to be replaced with Doctrine entity when right management
+        // won't be directly linked to the tab creation
+        // @ToDo
+        $tab = new Tab();
+        $tab->active = $data->getBoolean('visible', false);
```

Comment:
The tabs explicitly declared by the module developer are most likely to be displayed. This property needs to be set to `true`